### PR TITLE
add reference to docker hub credentials + cleanup

### PR DESCRIPTION
**Pull request by williamfalconeruk**
## Purpose
Add Dockerhub authentication to CircleCI pipeline. See Learning notes and ticket for details.

Fixes LPAL-170

## Approach

Add details to retrieve env vars with docker credentials, to service authenticated pulls.

## Learning

We have prerequisite technical guidance, to cover set up steps for this: https://ministryofjustice.github.io/opg-technical-guidance/runbooks/CI_pipelines/adding-ci-dockerhub-users/#adding-a-service-dockerid-to-circleci-build, which we hope to improve in the future.
 see also: 
- Using Docker Authenticated Pulls https://circleci.com/docs/2.0/private-images/ 

## Checklist

* [x] I have performed a self-review of my own code
* [ ] I have updated documentation (Confluence/GitHub wiki/tech debt doc) where relevant
* [ ] I have added tests to prove my work
* [ ] The product team have tested these changes

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -213,10 +211,17 @@ workflows:
 orbs:
   slack: circleci/slack@3.3.0
   build_containers_and_push_to_ecr:
+    orbs:
+      docker: circleci/docker@1.4.0
     commands:
+      dockerhub_login:
```
Comment (charlesmmarshall) [inline .circleci/config.yml:217]: Do the DOCKER_USER etc in this step need to be vars (so $)?

Comment (williamfalconeruk) [reply .circleci/config.yml:217]: The docker circle CI orb recognises the values as environment var names so we just have to pass that and it works it out. see: https://circleci.com/developer/orbs/orb/circleci/docker#commands-check  (expand the panel to see the code) and https://circleci.com/docs/2.0/reusing-config/#environment-variable-name

Comment (williamfalconeruk) [reply .circleci/config.yml:217]: the others do need the login details with a $ as per the docker auth pull docs in circle ci: https://circleci.com/docs/2.0/private-images/

**Review by Raffers — APPROVED** [review]
lgtm

**Pull request merged**